### ISSUE TITLE
AccessibilityInfo: include modality information

### ISSUE THREAD

**Issue by necolas** [necolas/react-native-web#1111]
**Is your feature request related to a problem? Please describe.**

When an app supports multiple modality types, e.g., touch, mouse, and keyboard, the way that focus is handled for each modality may differ. For example, on the web it is common to want a visible focus ring around elements that have received focus following a keyboard event (i.e., tabbing through focusable elements) or programmatic movement of focus (i.e., moving focus to a button in a modal).

RNW currently automatically handles show/hide of the default focus ring based on modality, but it doesn't have a way to surface which modality is considered active or whether a focus ring is visible. There are UI patterns where you may want to customise the "focus ring" by removing the default web outline and instead adding shadows or borders, etc., but it's not possible to do this for keyboard-only interactions because you don't know if the focus was triggered by a keyboard event. Other things you might want to do: lazy-load keyboard shortcut code or UI when the keyboard becomes active outside of a textinput; change/remove the hitslop from touchables when the mouse is active; enable hover states when the mouse is active

(These issues may also manifest in RN for native apps that can be controlled using a mouse and keyboard)

**Describe a solution you'd like**

Surface the state of the Modality helper to users of RNW. Possibly hang it off the existing `AccessibilityInfo` API from React Native and fire events when modality changes. It already has iOS-only features and events, so adding web-only (for now) seems ok. Will discuss with the RN team.

**Describe alternatives you've considered**

Alternatively, a new web-only API could be added.

**Comment by necolas** [comment]
Closing this. I developed a hook for this in another library and we'll figure out how to introduce a similar API to React Native at some point.
